### PR TITLE
Exclude fee=no for AddAcceptsCards

### PR DESCRIPTION
**Pull request by HolgerJeromin**
I found this:
https://www.openstreetmap.org/node/7670739053
Even if this a special case.

Edit:
Found more fishy elements
https://www.openstreetmap.org/node/757197561
https://www.openstreetmap.org/way/1214599523
https://www.openstreetmap.org/node/338438554

### CODE REVIEW

**Comment by matkoniecz** [comment]
This is asked for `amenity ~ restaurant|cafe|fast_food|ice_cream|food_court|pub|bar` so would not be asked for https://www.openstreetmap.org/node/7670739053

would not be asked also for https://www.openstreetmap.org/node/757197561 and https://www.openstreetmap.org/way/1214599523

fee=no for https://www.openstreetmap.org/node/338438554 was added by non-sc editor in edit titled "Specify toilet fees"  so seems clearly spurious

is there any  `amenity ~ restaurant|cafe|fast_food|ice_cream|food_court|pub|bar`  where not only `fee=no` would be applicable but also there would be no possibility of even voluntary payment?

**Comment by HolgerJeromin** [comment]
> This is asked for amenity ~ restaurant|cafe|fast_food|ice_cream|food_court|pub|bar so would not be asked for https://www.openstreetmap.org/node/7670739053

You overlooked:

```
or (shop and shop !~ no|vacant|mall)
```
This "laundry" was really asked a few hours ago...

The other objects were found via overpass turbo.

**Review by matkoniecz — APPROVED** [review]

**Comment by matkoniecz** [comment]
I also repaired some clearly broken `fee=no` - including some reported here. See say https://www.openstreetmap.org/changeset/154606600

File: app/src/main/java/de/westnordost/streetcomplete/quests/accepts_cards/AddAcceptsCards.kt [hunk 1]
```diff
@@ -20,6 +20,7 @@ class AddAcceptsCards : OsmFilterQuestType<CardAcceptance>() {
         and !payment:credit_cards and !payment:debit_cards and payment:others != no
         and !brand and !wikipedia:brand and !wikidata:brand
         and (!seasonal or seasonal = no)
+        and (!fee or fee = no)
         and (name or noname = yes or name:signed = no)
```
Comment (HolgerJeromin) [inline app/src/main/java/de/westnordost/streetcomplete/quests/accepts_cards/AddAcceptsCards.kt:24]: My check was wrong?!

**Comment by matkoniecz** [comment]
Oh, and have you tested this code? In emulator or on device?

(I have not done it so far)

**Comment by HolgerJeromin** [comment]
> Oh, and have you tested this code? In emulator or on device?

Tested on device with
https://github.com/HolgerJeromin/StreetComplete/actions/runs/10169947929
The laundry is not asked anymore.
This garden centre shop is still asked:
https://www.openstreetmap.org/way/37757063

**Pull request merged**